FIPS/fip-atomic_fvm_switch.md

```diff
@@ -0,0 +1,203 @@
+---
+fip: <to be assigned>
+title: Atomic switch to non-programmable FVM
+authors: Raúl Kripalani (@raulk), Steven Allen (@stebalien)
+discussions-to: 
+status: Draft
+type: Technical Core
+category: Core
+created: 2022-02-03
+spec-sections: 
+  - TBD
+requires:
+  - 
+replaces: N/A
+---
+
+# Atomic switch to non-programmable FVM
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+
+- [Simple Summary](#simple-summary)
+- [Abstract](#abstract)
+- [Change Motivation](#change-motivation)
+- [Specification](#specification)
+  - [WASM bytecode management](#wasm-bytecode-management)
+  - [Warming up the module cache](#warming-up-the-module-cache)
+  - [Atomic switch of execution layer](#atomic-switch-of-execution-layer)
+  - [Migration procedure](#migration-procedure)
+  - [Optional removal of intrinsic VM from Filecoin client codebases](#optional-removal-of-intrinsic-vm-from-filecoin-client-codebases)
+- [Design Rationale](#design-rationale)
+- [Backwards Compatibility](#backwards-compatibility)
+- [Test Cases](#test-cases)
+- [Security Considerations](#security-considerations)
+- [Incentive Considerations](#incentive-considerations)
+- [Product Considerations](#product-considerations)
+- [Implementation](#implementation)
+- [Copyright](#copyright)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Simple Summary
+
+This FIP motions for an **atomic switch** from the current intrinsic VM to a
+**non-programmable version of the Filecoin Virtual Machine**, at a chain epoch
+TBD.
+
+A baseline specification of the Filecoin Virtual Machine is provided in
+[filecoin-project/FIPs#288](https://github.com/filecoin-project/FIPs/pull/288).
+Only the canonical built-in actors will be supported after the atomic switch.
+
+This switch won't introduce new user-facing features or capabilities, but it's
+an important step in the trajectory towards full on-chain user-programmability.
```

Comment:
Well, I'd say it should also improve network security by ensuring that execution is deterministic and sandboxed, but I'm not sure if that counts as "user facing".